doc/mgmt/SONiC_Management_Framework_platform.md

```diff
@@ -0,0 +1,439 @@
+        |  +--rw config
+        |  +--ro state
+        |     +--ro oc-pf-ext:direction?         string
+        |     +--ro oc-pf-ext:speed-tolerance?   uint32
+        |     +--ro oc-pf-ext:target-speed?      uint32
+        |     +--ro oc-fan:speed?                uint32
+```
+
+### 3.6.2 CLI
+There are only show commands for platform information.
+#### 3.6.2.1 Configuration Commands
+#### 3.6.2.2 Show Commands
+```
+sonic# show platform
+  environment  Show platform Environment
+  fanstatus    Show platform fan status
+  psustatus    Show platform PSU status
+  psusummary   Show platform PSU summary
+  syseeprom    Show platform EEPROM information
+  temperature  Show platform temperature sensors
+  |            Pipe through a command
+  <cr>
+```
+##### Show EERPOM
+```
+sonic# show platform syseeprom
+-----------------------------------------------------------
+Attribute            Value/State
+-----------------------------------------------------------
+description         :x86_64-dell_z9100_c2538-r0
+empty               :False
+hardware-version    :A02
+id                  :Z9100-ON
+location            :Slot 1
+mfg-date            :2017-08-07
+mfg-name            :CES00
+name                :System Eeprom
+base-mac-address    :34:17:EB:2C:D7:00
+diag-version        :3.23.4.1
+mac-addresses       :384
+manufacture-country :CN
+onie-version        :3.23.1.3
+service-tag         :847RG02
+vendor-name         :DELL
+oper-status         :ACTIVE
+part-no             :0KY5C4
+removable           :False
+serial-no           :CN0KY5C4CES007770009
+software-version    :'dell_sonic_3.x_share.129-6a2cad56d'
+
+```
+
+##### Show  PSU status
+```
+sonic# show platform psustatus
+-----------------------------------------------------------
+PSU                  Status
+-----------------------------------------------------------
+PSU 1                ACTIVE
+PSU 2                INACTIVE
+
+```
+##### Show PSU summary
+```
+sonic# show platform psusummary
+PSU 1:
+    description          02RPHX
+    mfg-name             DELL
+    fans                 1
+    status-led           green
+    oper-status          ACTIVE
+    serial-no            CN-02RPHX-17972-74S-018N-A00
+    output-current (W)   1.27
+    output-power (W)     127.50
+    output-voltage (V)   12.19
+    fan-speed (RPM)      5808
+    fan-direction        fan_direction_exhaust
+PSU 2:
+    description          02RPHX
+    mfg-name             DELL
+    fans                 1
+    status-led           off
+    oper-status          INACTIVE
+    serial-no            CN-02RPHX-17972-74I-0139-A00
+    fan-speed (RPM)      0
+    fan-direction        fan_direction_exhaust
+```
+##### Show temperature
+```
+sonic# show platform temperature
+TH - Threshold
+--------------------------------------------------------------------------------------------------------------
+Name                     Temperature    High TH        Low TH         Critical High TH         Critical Low TH
```

Comment:
Hi Garrick,

I am porting the community's 'show platform temperature' CLICK cli in braodcom_sonic_3.x and I see that they are also printing 'Warning' and 'Timestamp'. 

header = ['Sensor', 'Temperature', 'High TH', 'Low TH', 'Crit High TH', 'Crit Low TH', **'Warning', 'Timestamp'**]

extra keys used are highlighted below,

9) **"warning_status"**
10) "False"
11) "critical_high_threshold"
12) "98.0"
13) "critical_low_threshold"
14) "N/A"
15) **"timestamp"**
16) "20200609 15:33:41"

If we print these two keys in the o/p, it will be in-line with the community implemented CLI.

Reply:
I can add that in.

Reply:
Yes. Please add it.

Reply:
@fk410167 - Okay, so I took a look and I will be able to add the warning status. But the timestamp, you'll need to change the format in the dB so it matches this pattern specified in `ietf-yang-types.yang`

Please follow the pattern specified by this type:

```
  typedef date-and-time {
    type string {
      pattern '\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?'
            + '(Z|[\+\-]\d{2}:\d{2})';
    }
    description
     "The date-and-time type is a profile of the ISO 8601
      standard for representation of dates and times using the
      Gregorian calendar.  The profile is defined by the
      date-time production in Section 5.6 of RFC 3339.

      The date-and-time type is compatible with the dateTime XML
      schema type with the following notable exceptions:

      (a) The date-and-time type does not allow negative years.

      (b) The date-and-time time-offset -00:00 indicates an unknown
          time zone (see RFC 3339) while -00:00 and +00:00 and Z
          all represent the same time zone in dateTime.

      (c) The canonical format (see below) of data-and-time values
          differs from the canonical format used by the dateTime XML
          schema type, which requires all times to be in UTC using
          the time-offset 'Z'.

      This type is not equivalent to the DateAndTime textual
      convention of the SMIv2 since RFC 3339 uses a different
      separator between full-date and full-time and provides
      higher resolution of time-secfrac.

      The canonical format for date-and-time values with a known time
      zone uses a numeric time zone offset that is calculated using
      the device's configured known offset to UTC time.  A change of
      the device's offset to UTC time will cause date-and-time values
      to change accordingly.  Such changes might happen periodically
      in case a server follows automatically daylight saving time
      (DST) time zone offset changes.  The canonical format for
      date-and-time values with an unknown time zone (usually
      referring to the notion of local time) uses the time-offset
      -00:00.";
    reference
     "RFC 3339: Date and Time on the Internet: Timestamps
      RFC 2579: Textual Conventions for SMIv2
      XSD-TYPES: XML Schema Part 2: Datatypes Second Edition";
  }

```

Reply:
Hi Garrick,

Cant you just keep it as string? 
In the community HLD which talks about these fields, I see it is taken as 'string'. 

I actually made the code changes for KLISH, and tested locally. They seems to be working. 'warning_status' would be of type boolean where as 'timestamp' would be of the type string.

My code changes are under review as of now. One they are pushed, you could check them in dell code drop.

-Fuzail

Reply:
I meant keep it as string without following specific pattern. If we go for changing the pattern of the 'timestamp' field, we will be digressing from community code.

Reply:
@fk410167 - it's fine. I'll convert the string in the dB to be compatible with the YANG's pattern. The timestamp:
`20200707 21:53:10`

will always have a format like this in the db right?
format: `YYYYDDMM HH:MM:SS`

Reply:
Yes, the format will be like this only.

Before making the changes, please wait to check the changes I did. You will receive the changes in Dell code drop in couple of days. If you feel there is a need of conversion, you can do it then.

Thanks,

Reply:
@fk410167 - what changes have you made? Does it affect the format of the timestamp?

Reply:
No, just added the below leafs in the openconfig-platform-ext.yang.
  
      leaf warning-status {
            type boolean;
            description
                "Indication for temperature warning status";
        }

        leaf timestamp {
            type string;
            description "The timestamp for the fetched temperature";
        }

And in the backend, read the timestamp string from the db and store (& display) it as is.

Reply:
@fk410167 - So this code has been merged already? You could of at least given me a heads-up so I don't have to spend time on it since you told me to work on it. If the code hasn't been merged, send me an email because I have a .diff that uses the proper type for the timestamp and also I have the conversion function that converts to the proper OC Yang format for the timestamp.